File: Graph Representation/C++/graph_adj_matrix.cpp
#include <iostream>
#include <vector>

class GraphAdjMatrix {
private:
    std::vector<std::vector<int>> adjMatrix;
    int numVertices;

public:
    GraphAdjMatrix(int numVertices) {
        this->numVertices = numVertices;
        adjMatrix.resize(numVertices, std::vector<int>(numVertices, 0));
    }

    void addEdge(int i, int j) {
        adjMatrix[i][j] = 1;
        adjMatrix[j][i] = 1; // For undirected graph
    }

    void printGraph() {
        for (int i = 0; i < numVertices; i++) {
            for (int j = 0; j < numVertices; j++) {
                std::cout << adjMatrix[i][j] << " ";
            }
            std::cout << std::endl;
        }
    }
};

int main() {
    GraphAdjMatrix graph(6);
    graph.addEdge(0, 1);
    graph.addEdge(0, 2);
    graph.addEdge(1, 2);
    graph.addEdge(2, 3);
    graph.addEdge(4, 5);

    graph.printGraph();
    return 0;
}

/*
*   >>> OUTPUT <<<
*
*  0 1 1 0 0 0 
*  1 0 1 0 0 0 
*  1 1 0 1 0 0 
*  0 0 1 0 0 0 
*  0 0 0 0 0 1 
*  0 0 0 0 1 0 
*/

File: Graph Representation/Java/GraphAdjMatrix.java
public class GraphAdjMatrix {
    private int[][] adjMatrix;
    private int numVertices;

    public GraphAdjMatrix(int numVertices) {
        this.numVertices = numVertices;
        adjMatrix = new int[numVertices][numVertices];
    }

    public void addEdge(int i, int j) {
        adjMatrix[i][j] = 1;
        adjMatrix[j][i] = 1; // For undirected graph
    }

    public void printGraph() {
        for (int i = 0; i < numVertices; i++) {
            for (int j = 0; j < numVertices; j++) {
                System.out.print(adjMatrix[i][j] + " ");
            }
            System.out.println();
        }
    }

    public static void main(String[] args) {
        GraphAdjMatrix graph = new GraphAdjMatrix(6);
        graph.addEdge(0, 1);
        graph.addEdge(0, 2);
        graph.addEdge(1, 2);
        graph.addEdge(2, 3);
        graph.addEdge(4, 5);

        graph.printGraph();
    }
}
